### PR TITLE
Fix optional fields in Pydantic models

### PR DESCRIPTION
**Pull request by orkhank**
## Problem

[Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) is flagging initialization of some [Pydantic](https://docs.pydantic.dev/latest/) objects (ie. objects of classes extending [`BaseModel`](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel)) as having missing arguments when optional arguments are not provided. Although the code still runs as it should, this type of error is a potential source of frustration for VS Code users.

### Example

For example, according to the doc string for the [`GoogleGenerativeAIEmbeddings`](https://api.python.langchain.com/en/latest/embeddings/langchain_google_genai.embeddings.GoogleGenerativeAIEmbeddings.html) class, it can be used in the following way:

```python
from langchain_google_genai import GoogleGenerativeAIEmbeddings

embeddings = GoogleGenerativeAIEmbeddings(model="models/embedding-001")
embeddings.embed_query("What's our Q1 revenue?")
```

When editing the same code in Visual Studio Code with the Pylance extension enabled, following error is encountered.
image

In this case, the parameters `task_type`, `google_api_key`, `client_options`, `transport`, and `request_options` are clearly optional, as indicated by the `None` type. `client` does not have the `None` type, but is also optional since it is not provided in the example code snippet.

Note that Pylance does not flag the absence of the `credentials` field, even though it is also optional, and that the only field with a _recognized_ default value (`= None`) is `credentials`.

## Causes

Upon closer examination of the code, it becomes evident that some definitions for Pydantic fields are not handled by Pylance as intended. Examining the definition of [`GoogleGenerativeAIEmbeddings`](https://api.python.langchain.com/en/latest/embeddings/langchain_google_genai.embeddings.GoogleGenerativeAIEmbeddings.html) from the previous example (some methods, field descriptions and doc string for class are redacted for the sake of brevity):

```python
class GoogleGenerativeAIEmbeddings(BaseModel, Embeddings):
    # <doc string>

    client: Any  #: :meta private:
    model: str = Field(
        ...,
        description="...",
    )
    task_type: Optional[str] = Field(
        None,
        description="...",
    )
    google_api_key: Optional[SecretStr] = Field(
        None,
        description="...",
    )
    credentials: Any = Field(
        default=None,  # <- notice the default keyword
        exclude=True,
        description="...",
    )
    client_options: Optional[Dict] = Field(
        None,
        description="...",
    )
    transport: Optional[str] = Field(
        None,
        description="...",
    )
    request_options: Optional[Dict] = Field(
        None,
        description="...",
    )

    @root_validator()
    def validate_environment(cls, values: Dict) -> Dict:
        """Validates params and passes them to google-generativeai package."""
        google_api_key = get_from_dict_or_env(
            values, "google_api_key", "GOOGLE_API_KEY"
        )
        client_info = get_client_info("GoogleGenerativeAIEmbeddings")

        values["client"] = build_generative_service(
            credentials=values.get("credentials"),
            api_key=google_api_key,
            client_info=client_info,
            client_options=values.get("client_options"),
        )
        return values


    # <other methods>
```

we can see that except for `client` and `model` all other fields are given a default value by [`Field`](https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.Field). However, the only default value recognized by Pylance is the one for `credentials`. And it is no coincidence that `credentials` is also the only field where the default value is defined using the `default` **keyword** argument.

It is safe to say that the errors related to `task_type`, `google_api_key`, `client_options`, `transport`, and `request_options` are caused by the default value for them not being provided with the keyword arguments. (Unfortunately, I am not sure why this is the case.)

For the `client` case, we can assume that it's default value is `None`, since it's set within the `validate_environment` method. This case is not explicitly declared, which makes Pylance angry.

### TL;DR
The problem seems to be caused by two different cases:

1. The default value for an optional field is not provided (such as `client` in [`GoogleGenerativeAIEmbeddings`](https://api.python.langchain.com/en/latest/embeddings/langchain_google_genai.embeddings.GoogleGenerativeAIEmbeddings.html)).
2. The default value for an optional field is provided without the `default` keyword using `Field`.

## Solution

Find all the field definitions that are causing problems and fix them. 

To find the examples for the first case, I searched for field definitions of type `Any` without a default value, similar to `client` in [`GoogleGenerativeAIEmbeddings`](https://api.python.langchain.com/en/latest/embeddings/langchain_google_genai.embeddings.GoogleGenerativeAIEmbeddings.html). For the second case, I searched for optional fields using the `Optional[.*] = Field(` regex and manually checked for definitions without the `default` keyword.

It was trivial to apply the fix after finding the faulty lines.

### CODE REVIEW

**Comment by lkuligin** [comment]
@orkhank thanks for your contribution! could fix the remaining linter issues, please?

**Review by lkuligin — APPROVED** [review]
